mon-pix/app/routes/campaigns/start-or-resume.js

```diff
@@ -16,98 +16,19 @@ export default class StartOrResumeRoute extends Route.extend(SecuredRouteMixin)
-    return this.transitionTo('campaigns.restricted.login-or-register-to-access', campaign.code, { queryParams: { displayRegisterForm } });
-  }
-
-  get _shouldVisitPoleEmploiLoginPage() {
-    return this.state.isCampaignPoleEmploi && !this.state.isUserLoggedInPoleEmploi;
-  }
-
-  async model() {
-    this.isLoading = true;
-    return this.modelFor('campaigns');
-  }
-
-  async redirect(campaign) {
-    if (campaign.isArchived) {
-      this.isLoading = false;
-      return;
-    }
-
-    const ongoingCampaignParticipation = await this.store.queryRecord('campaignParticipation', { campaignId: campaign.id, userId: this.currentUser.user.id });
-    this._updateStateFrom({ ongoingCampaignParticipation });
-
-    if (this._shouldVisitLandingPageAsLoggedUser) {
-      return this.replaceWith('campaigns.campaign-landing-page', campaign);
-    }
-
-    if (this._shouldProvideExternalIdToAccessCampaign) {
-      return this.replaceWith('campaigns.fill-in-participant-external-id', campaign);
-    }
-
-    if (this._shouldStartCampaignParticipation) {
-      try {
-        const campaignParticipation = await this.store.createRecord('campaign-participation', { campaign, participantExternalId: this.state.participantExternalId }).save();
-        this._updateStateFrom({ ongoingCampaignParticipation: campaignParticipation });
-      } catch (err) {
-        if (get(err, 'errors[0].status') === 403) {
-          this.session.invalidate();
-          return this.transitionTo('campaigns.start-or-resume', campaign);
-        }
-        return this.send('error', err, this.transitionTo('campaigns.start-or-resume'));
-      }
-    }
-
-    if (campaign.isProfilesCollection) {
-      return this.replaceWith('campaigns.profiles-collection.start-or-resume', campaign);
-    } else {
-      return this.replaceWith('campaigns.assessment.start-or-resume', campaign);
-    }
-  }
-
-  get _shouldJoinRestrictedCampaign() {
-    return this.state.isCampaignRestricted
-      && (this.state.isUserLogged || this.state.externalUser)
-      && !this.state.hasUserCompletedRestrictedCampaignAssociation;
-  }
-
-  get _shouldJoinSimplifiedCampaignAsAnonymous() {
-    return this.state.hasUserSeenLandingPage
-      && this.state.isCampaignSimplifiedAccess
-      && !this.state.isUserLogged;
-  }
-
-  _shouldResetState(campaignCode) {
-    return campaignCode !== this.state.campaignCode;
-  }
-
-  _redirectToPoleEmploiLoginPage(transition) {
-    this.session.set('attemptedTransition', transition);
-    return this.transitionTo('login-pe');
-  }
-
   async beforeModel(transition) {
     this.authenticationRoute = 'inscription';
     const campaign = this.modelFor('campaigns');
     if (this._shouldResetState(campaign.code)) {
       this._resetState();
     }
+
     this._updateStateFrom({ campaign, queryParams: transition.to.queryParams, session: this.session });
 
+    if (this.state.isCampaignForNoviceUser) {
+      this.state.hasUserSeenLandingPage = true;
+    }
```

Comment:
**proposition:**
Déplacer cette logique dans `this._updateStateFrom`

A la place de : 
```
    const hasUserSeenLandingPage = this._handleQueryParamBoolean(queryParams.hasUserSeenLandingPage, this.state.hasUserSeenLandingPage);
```
mettre
```
const isCampaignForNoviceUser = get(campaign, 'isForAbsoluteNovice', this.state.isCampaignForNoviceUser)
    const hasUserSeenLandingPage = isCampaignForNoviceUser ? true : this._handleQueryParamBoolean(queryParams.hasUserSeenLandingPage, this.state.hasUserSeenLandingPage);

```

Reply:
Bien vu !

Reply:
Je reprends la réponse de @MelanieMEB, mais je suggère de ne pas modifier `hasUserSeenLandingPage` mais plutôt de rajouter une variable dans le state:

```
const shouldDisplayLandingPage = isCampaignForNoviceUser ? true : this._handleQueryParamBoolean(queryParams.hasUserSeenLandingPage, this.state.shouldDisplayLandingPage);

this.state = {
...
  shouldDisplayLandingPage,
...
}
```

Dans le `resetState`:
```
this.state = {
  shouldDisplayLandingPage: false;
}
```

Cela permet ensuite de **supprimer** `hasUserSeenLandingPage` du state. Et partout où on a besoin de `hasUserSeenLandingPage`, on utilise `shouldDisplayLandingPage` à la place.

Reply:
Entièrement ok pour ce changement, mais on est d'accord pour initialiser `shouldDisplayLandingPage` à `true` de base dans le state plutôt que `false` non ?
Qui du coup inversera tous les appels à l'ancien `hasUserSeenLandingPage ` dans les conditions.

Reply:
Oui, l'initialiser à `false`. Je modifie ma réponse précédente.